### PR TITLE
openalSoft: 1.21.1 -> 1.22.0

### PR DESCRIPTION
**Pull request by fufexan**
###### Description of changes

Update to new version.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jtojnar** [comment]
How did you determine the patch is obsolete?

**Comment by jtojnar** [comment]
Looks like the file was just moved:

**Comment by fufexan** [comment]
> How did you determine the patch is obsolete?

I tried just changing the version, and the build failed, saying there's no such file to patch. I tried grepping for the relevant info in the failed build dir but found no more occurrences of `@OUT@`.

I've tried running the binaries and they worked fine. However I haven't tested the rest of the dependents (nixpkgs-review closure was about 60G).

I will try to fix it if the patch is still needed.

**Comment by jtojnar** [comment]
`@OUT@` was only introduced by the patch you removed. See the [original commit](https://github.com/NixOS/nixpkgs/commit/31cf1ff4ee6d7181a679f1d5bcd7a5d82e6f7c5e) for how to test it.

**Comment by ilya-fedin** [comment]
> See the [original commit](https://github.com/NixOS/nixpkgs/commit/31cf1ff4ee6d7181a679f1d5bcd7a5d82e6f7c5e) for how to test it.

:thinking: I wonder how it was working all the time given that hrtf files are in openal/hrtf and the patch doesn't add the subdir argument to the path. Also, I wonder how the NIX_LDFLAGS works given that it will load the symbols from dlopen unless you specify `-DALSOFT_DLOPEN=OFF`. And, finally, this PR doesn't add pipewire to the buildInputs :P

**Comment by ilya-fedin** [comment]
It also has dbus dependency now for rtkit integration

**Comment by fufexan** [comment]
> And, finally, this PR doesn't add pipewire to the buildInputs :P

Whoops, my bad xD

> It also has dbus dependency now for rtkit integration

Willfix

**Comment by fufexan** [comment]
Btw @ilya-fedin should I include https://github.com/kcat/openal-soft/commit/2ea5bb91686ee1a7350672bf15b7b63649dbf111 as a patch?

**Comment by ilya-fedin** [comment]
> Btw @ilya-fedin should I include [kcat/openal-soft@2ea5bb9](https://github.com/kcat/openal-soft/commit/2ea5bb91686ee1a7350672bf15b7b63649dbf111) as a patch?

No, it's for CentOS 7 :D
I try to upstream the NixOS patch though: https://github.com/kcat/openal-soft/pull/693

**Comment by ilya-fedin** [comment]
> Also, I wonder how the NIX_LDFLAGS workaround works

Ah, it makes nix write the RPATH to those libraries and then dlopen find them thanks to that :D

**Comment by fufexan** [comment]
Added the new deps. I can't really tell if it's building with dbus as it only shows up once in the logs (building wrapper_dbus.o or something).

**Comment by ilya-fedin** [comment]
It says about dbus during configure only if it can't find, so I think it found dbus. Can you replace NIX_LDFLAGS with `cmakeFlags = [ "-DALSOFT_DLOPEN=OFF" ];` so it actually would be able to use dbus/pipewire?

File: pkgs/development/libraries/openal-soft/default.nix [hunk 1]
```diff
@@ -1,40 +1,42 @@
...
     ++ lib.optional alsaSupport alsa-lib
+    ++ lib.optional dbusSupport dbus
+    ++ lib.optional pipewireSupport pipewire
     ++ lib.optional pulseSupport libpulseaudio
     ++ lib.optionals stdenv.isDarwin [ CoreServices AudioUnit AudioToolbox ];
 
-  NIX_LDFLAGS = toString (
-    lib.optional alsaSupport "-lasound"
-    ++ lib.optional pulseSupport "-lpulse");
+  cmakeFlags = [ "-DALSOFT_DLOPEN=OFF" ];
```
Comment (jtojnar) [inline pkgs/development/libraries/openal-soft/default.nix:39]:
```suggestion
  cmakeFlags = [
    # Would be nice to add a comment what this is for.
    "-DALSOFT_DLOPEN=OFF"
  ];
```

and also include an overview of the changes in the commit message.

Comment (fufexan) [reply pkgs/development/libraries/openal-soft/default.nix]: Done, I hope the message is clear enough.

**Comment by ilya-fedin** [comment]
Are we still waiting for something? :thinking:

**Review by jtojnar — APPROVED** [review]
Changelog looks okay to me: https://github.com/kcat/openal-soft/blob/1.22.0/ChangeLog

**Comment by Mindavi** [comment]
Oh btw, this broke cross-compilation for openal-soft, possibly due to some issues in pipewire:

```
/nix/store/npg4vm4f0fzpy1lj24ww2l6sqqksrk4r-pipewire-aarch64-unknown-linux-gnu-0.3.51-dev/include/spa-0.2/spa/utils/json.h: In function 'char* spa_json_format_float(char*, int, float)':
/nix/store/npg4vm4f0fzpy1lj24ww2l6sqqksrk4r-pipewire-aarch64-unknown-linux-gnu-0.3.51-dev/include/spa-0.2/spa/utils/json.h:263:20: error: 'isnormal' was not declared in this scope; did you mean 'std::isnormal'?
  263 |  if (SPA_UNLIKELY(!isnormal(val))) {
      |                    ^~~~~~~~
/nix/store/npg4vm4f0fzpy1lj24ww2l6sqqksrk4r-pipewire-aarch64-unknown-linux-gnu-0.3.51-dev/include/spa-0.2/spa/utils/defs.h:244:46: note: in definition of macro 'SPA_UNLIKELY'
  244 | #define SPA_UNLIKELY(x) (__builtin_expect(!!(x),0))
```

I'll probably look into this later, but just wanted to note it here. Not sure what the solution should be and why it's happening.

**Comment by Artturin** [comment]
> Oh btw, this broke cross-compilation for openal-soft, possibly due to some issues in pipewire:
> 
> ```
> /nix/store/npg4vm4f0fzpy1lj24ww2l6sqqksrk4r-pipewire-aarch64-unknown-linux-gnu-0.3.51-dev/include/spa-0.2/spa/utils/json.h: In function 'char* spa_json_format_float(char*, int, float)':
> /nix/store/npg4vm4f0fzpy1lj24ww2l6sqqksrk4r-pipewire-aarch64-unknown-linux-gnu-0.3.51-dev/include/spa-0.2/spa/utils/json.h:263:20: error: 'isnormal' was not declared in this scope; did you mean 'std::isnormal'?
>   263 |  if (SPA_UNLIKELY(!isnormal(val))) {
>       |                    ^~~~~~~~
> /nix/store/npg4vm4f0fzpy1lj24ww2l6sqqksrk4r-pipewire-aarch64-unknown-linux-gnu-0.3.51-dev/include/spa-0.2/spa/utils/defs.h:244:46: note: in definition of macro 'SPA_UNLIKELY'
>   244 | #define SPA_UNLIKELY(x) (__builtin_expect(!!(x),0))
> ```
> 
> I'll probably look into this later, but just wanted to note it here. Not sure what the solution should be and why it's happening.

the isnormal was added in https://gitlab.freedesktop.org/pipewire/pipewire/-/commit/2b16df4e899d7faf7611e228a855dc62b5bdee83 and adding support for pipewire in openal-soft just exposed it

let's continue in https://matrix.to/#/#cross-compiling:nixos.org

**Pull request merged**